server/oauth.ts

```diff
@@ -346,12 +347,27 @@ export const setLocalStateFromIdTokenOrUserCookie = (
 	// but not the other fields. This will allow the frontend to show the
 	// signed in menu, but not show the user's name or email.
 	const hasIdTokenOrUserCookie = idToken || req.cookies['GU_U'];
-	res.locals.identity = {
+
+	let identityIdClaim: string | undefined;
+	let nameClaim: string | undefined;
+	let emailClaim: string | undefined;
+
+	if (typeof idToken?.claims.legacy_identity_id === 'string') {
+		identityIdClaim = idToken?.claims.legacy_identity_id;
+	}
+	if (typeof idToken?.claims.name === 'string') {
+		nameClaim = idToken?.claims.name;
+	}
+	if (typeof idToken?.claims.email === 'string') {
+		emailClaim = idToken?.claims.email;
+	}
```

Comment:
all of this code is jut because the claims are declared as type `unknown`, I don't know if there's a better way to just convert them to  optional strings